cmds/core/netcat/netcat.go

```diff
@@ -16,59 +16,106 @@ import (
 	"github.com/u-root/u-root/pkg/uroot/util"
 )
 
-const usage = "netcat [go-style network address]"
+const usage = "netcat host port"
 
-var (
-	netType = flag.String("net", "tcp", "What net type to use, e.g. tcp, unix, etc.")
-	listen  = flag.Bool("l", false, "Listen for connections.")
-	verbose = flag.Bool("v", false, "Verbose output.")
-)
+var errMissingHostnameOrPort = fmt.Errorf("missing hostname or port")
+
+type params struct {
+	network string
+	listen  bool
+	verbose bool
+	host    string
+	port    string
+}
+
+func parseParams() params {
+	netType := flag.String("net", "tcp", "What net type to use, e.g. tcp, unix, etc.")
+	listen := flag.Bool("l", false, "Listen for connections.")
+	verbose := flag.Bool("v", false, "Verbose output.")
+	flag.Parse()
+
+	return params{
+		network: *netType,
+		listen:  *listen,
+		verbose: *verbose,
+	}
+}
+
+type cmd struct {
+	stdin  io.Reader
+	stdout io.Writer
+	stderr io.Writer
+	addr   string
+	params
+}
+
+func command(stdin io.Reader, stdout io.Writer, stderr io.Writer, args []string, p params) (*cmd, error) {
+	if len(args) != 2 {
```

Comment:
what if there are 3 args? It's not a missing name in that case I guess? It's too many?

Reply:
Yeah, checking for != 2 was a mistake, interstingly that on linux `nc -l localhost 8080 xxx` will return error and print usage, on macOS it just ignore it. But if you prefer Go style I will revert it to previous style.